src/vm/checker/typecheck/tests/mod.rs

```diff
@@ -286,6 +286,55 @@ fn test_factorial() {
     type_check(&":transient:", &mut contract, &mut analysis_db, false).unwrap();
 }
 
+#[test]
+fn test_options() {
+    let contract = "
+         (define (foo (id (optional int)))
+           (+ 1 (default-to 1 id)))
+         (define (bar (x int))
+           (if (> 0 x)
+               (some x)
+               none))
+         (+ (foo none)
+            (foo (bar 1))
+            (foo (bar 0)))
+         ";
+
+    let mut contract = parse(contract).unwrap();
+    let mut analysis_conn = AnalysisDatabaseConnection::memory();
+    let mut analysis_db = analysis_conn.begin_save_point();
+
+    type_check(&":transient:", &mut contract, &mut analysis_db, false).unwrap();
```

Comment:
If we're testing 2 different contracts and 2 different kind of expectations, should we have 2 different tests?

Reply:
My inclination is usually only to do explicitly separate tests in cases like error conditions -- in this test, everything should be legal, so if anything fails, the test fails. It does make diagnosing _why_ a test failed a little harder, but I worry less about that, since that means there will have to be some debugging happening anyways.